File: Relaciones/ejercicio 1/Departamento.py
class Departamento:
    def __init__(self, a=None):
        self.n = [None] * 20
        if a is None:
            self.nombre = "A1"
            self.area = "social"
            self.nroEmpleados = 5
            self.n[0] = Empleado("Jhon")
            self.n[1] = Empleado()
            self.n[2] = Empleado("Arturo")
            self.n[3] = Empleado("Gael")
            self.n[4] = Empleado("Miguel")
        else:
            self.nombre = "A2"
            self.area = "familiar"
            self.nroEmpleados = a
            for i in range(self.nroEmpleados):
                self.n[i] = Empleado()

    def mostrarEmpleados(self):
        print(self.nombre, self.area, self.nroEmpleados, "empleados")
        if self.nroEmpleados == 0:
            print("Este departamento no tiene ningun empleado")
        for i in range(self.nroEmpleados):
            self.n[i].mostrar()

    def cambiarSalario(self, c):
        for i in range(self.nroEmpleados):
            self.n[i].setSueldo(c)

    def verificarSiPertenece(self, c):
        k = 0
        for i in range(self.nroEmpleados):
            for j in range(c.getNroEmpleados()):
                if self.n[i].getNombre() == c.n[j].getNombre():
                    print(self.n[i].getNombre(), "Pertenece a ambos departamentos")
                    k += 1
        if k == 0:
            print("Ningun empleado pertenece a ambos departamentos")

    def cambiarEmpleados(self, z):
        temp = self.n
        self.n = z.getN()
        z.setN(temp)
        aux = self.nroEmpleados
        self.nroEmpleados = z.getNroEmpleados()
        z.setNroEmpleados(aux)

    def setNombre(self, nombre):
        self.nombre = nombre

    def getArea(self):
        return self.area

    def setArea(self, area):
        self.area = area

    def getNroEmpleados(self):
        return self.nroEmpleados

    def setNroEmpleados(self, nroEmpleados):
        self.nroEmpleados = nroEmpleados

    def getN(self):
        return self.n

    def setN(self, n):
        self.n = n
if __name__ == "__main__":
    d1 = Departamento()
    d2 = Departamento(3)

    print("\n--- Departamento 1 ---")
    d1.mostrarEmpleados()

    print("\n--- Departamento 2 ---")
    d2.mostrarEmpleados()

    print("\n--- Cambiando salario d2 a 5000 ---")
    d2.cambiarSalario(5000)
    d2.mostrarEmpleados()

    print("\n--- Verificar si comparten empleados d1 y d2 ---")
    d1.verificarSiPertenece(d2)

    print("\n--- Intercambiando empleados d1 <-> d2 ---")
    d1.cambiarEmpleados(d2)

    print("\n--- Nuevos empleados d1 ---")
    d1.mostrarEmpleados()

    print("\n--- Nuevos empleados d2 ---")
    d2.mostrarEmpleados()

File: Relaciones/ejercicio 1/Departamento.java
package relaciones;
public class Departamento {
    private String nombre,area;
    private  int nroEmpleados;
    private Empleado n[]=new Empleado[20];
    
    
    public Departamento(){
        nombre="A1";
        area="social";
        nroEmpleados=5;
        n[0]=new Empleado("Jhon");
        n[1]=new Empleado();
        n[2]=new Empleado("Arturo");
        n[3]=new Empleado("Gael");
        n[4]=new Empleado("Miguel");
      
    }
    public Departamento(int a){
        nombre="A2";
        area="familiar";
        nroEmpleados=a;
         for(int i=0;i<nroEmpleados;i++){
             n[i]=new Empleado();
         }
    }
    
    public void mostrarEmpleados(){
        System.out.println(nombre+" "+area+" "+nroEmpleados+"  empleados");
        if (nroEmpleados==0){
            System.out.println("Este departamento no tiene ningun empleado");
        }
        for(int i=0;i<nroEmpleados;i++){
             n[i].mostrar();
         }
    }
    
    public void cambiarSalario( double c){
        for(int i=0;i<nroEmpleados;i++){
             n[i].setSueldo(c);
        }
    }
    
    public void verificarSiPertenece(Departamento c){
        int  k=0;
         for(int i=0;i<nroEmpleados;i++){
              for(int j=0;j<c.getNroEmpleados();j++){
                  if((n[i].getNombre()).equals(c.n[j].getNombre())){
                      System.out.println(n[i].getNombre()+"Pertenece a  ambos departamentos");
                      k++;
                  }
               }
         }
         if(k==0)
             System.out.println("Ningun empleado pertenece a ambos departamentos");
    }

   public void cambiarEmpleados(Departamento z) {
    Empleado[] temp = n;
    n = z.getN();
    z.setN(temp);
    int aux = nroEmpleados;
    nroEmpleados = z.getNroEmpleados();
    z.setNroEmpleados(aux);
}

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    public String getArea() {
        return area;
    }
    public void setArea(String area) {
        this.area = area;
    }
    public int getNroEmpleados() {
        return nroEmpleados;
    }
    public void setNroEmpleados(int nroEmpleados) {
        this.nroEmpleados = nroEmpleados;
    }
    public Empleado[] getN() {
        return n;
    }
    public void setN(Empleado[] n) {
        this.n = n;
    }
}

